### ISSUE TITLE
expect_comparison_linter() gets bad metadata for faulty expect_true() usage

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#2083]
Came across this:

```r
expect_true(is.count(n_draws), n_draws > 1)
```

This throws `expect_comparison_linter()`, but gives a bad message:

```r
lint(
  text = "expect_true(is.count(n_draws), n_draws > 1)",
  linters = expect_comparison_linter()
)
<text>:1:1: warning: [expect_comparison_linter] NA(x, y) is better than expect_true(x ( y).
expect_true(is.count(n_draws), n_draws > 1)
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Maybe we shouldn't invest in doing anything here since `expect_true()`'s been used incorrectly, but certainly it's not a helpful lint message.

**Comment by AshesITR** [comment]
Can the bug be triggered by a valid usage like

```r
expect_true(info = "my info first", a > b)
```
?

If so, I think putting a more robust XPath in place to find the comparator is a good idea.

**Comment by MichaelChirico** [comment]
> ```r
> expect_true(info = "my info first", a > b)
> ```

this doesn't lint, it's unusual enough that I would wait for a request first before supporting that.
